samcli/lib/iac/cdk/cdk_iac.py

```diff
@@ -31,3 +115,229 @@ def update_packaged_locations(self, stack: Stack) -> bool:
     @staticmethod
     def get_iac_file_patterns() -> List[str]:
         return ["cdk.json"]
+
+    def _cdk_synth(self, app: Optional[str] = None, context: Optional[List] = None) -> str:
+        """
+        Run cdk synth to get the cloud assembly
+        """
+        context = context or []
+        cdk_executable = _get_cdk_executable_path()
+        LOG.debug("CDK Toolkit found at %s", cdk_executable)
+        synth_command = [
+            cdk_executable,
+            "synth",
+            "--no-staging",
+        ]
+        if app is not None:
+            synth_command.append("--app")
+            synth_command.append(app)
+
+        if (app is None) or (app is not None and not os.path.isdir(app)):
+            synth_command.append("-o")
+            synth_command.append(self._cloud_assembly_dir)
+
+        if context is not None:
+            for keyvalpair in context:
+                synth_command.append("--context")
+                synth_command.append(keyvalpair)
+
+        try:
+            LOG.info("Synthesizing CDK App")
+            LOG.debug("command: %s", synth_command)
+            subprocess.check_output(
+                synth_command,
+                stderr=subprocess.STDOUT,
+            )
+        except subprocess.CalledProcessError as cdk_error:
+            output = cdk_error.output.decode("utf-8")
+            raise CdkSynthError(output) from cdk_error
+
+        if app is not None and os.path.isdir(app):
+            copy_tree(app, self._cloud_assembly_dir)
+
+        LOG.debug("Cloud assembly synthed at %s", self._cloud_assembly_dir)
+        return self._cloud_assembly_dir
+
+    def _get_project_from_cloud_assembly(self, cloud_assembly_dir: str) -> SamCliProject:
+        """
+        create a cdk project from cloud_assembly
+        """
+        cloud_assembly = CloudAssembly(cloud_assembly_dir, self._source_dir)
+        stacks: List[Stack] = [self._build_stack(cloud_assembly, ca_stack) for ca_stack in cloud_assembly.stacks]
+        project = SamCliProject(stacks=stacks)
+        return project
+
+    def _build_stack(
+        self, cloud_assembly: CloudAssembly, ca_stack: Union[CloudAssemblyStack, CloudAssemblyNestedStack]
+    ) -> Stack:
+        """
+        Extract stack from given CloudAssemblyStack
+        """
+        assets = _collect_assets(ca_stack)
+        LOG.debug("Found assets: %s", str(assets))
+        asset_parameters = {
+            asset_param
+            for asset in assets.values()
+            for asset_param in asset.extra_details.get("assetParameters", {}).values()
+        }
+        sections = {}
+        options_map = self._context.command_options_map
+        template = get_resolved_cfn_template(
```

Comment:
I think it's neat to do the intrinsics resolving here. But how do we get the original template back? That's  something we need to revisit when we start to work on the `write_project` implementation.

Reply:
I believe that topic is part of the planned discussion we have for `write_project`. There are a few options here, but I don't think this should be something difficult.

Reply:
sounds good. Just wanted to keep a note here as a reminder